src/fleet/azext_fleet/custom.py

```diff
@@ -316,10 +320,17 @@ def create_update_run(cmd,
                       stages=None,
                       update_strategy_name=None,
                       no_wait=False):
-    if upgrade_type == "Full" and kubernetes_version is None:
-        raise CLIError("Please set kubernetes version when upgrade type is 'Full'.")
-    if upgrade_type == "NodeImageOnly" and kubernetes_version is not None:
-        raise CLIError("Cannot set kubernetes version when upgrade type is 'NodeImageOnly'.")
+
+    if upgrade_type in UPGRADE_TYPE_ERROR_MESSAGES:
+        if (
+            ((upgrade_type == "Full" or upgrade_type == "ControlPlaneOnly") and kubernetes_version is None) or
```

Comment:
should check the constant value directly? UPGRADE_TYPE_FULL ...etc

Reply:
Yup, resolved!